data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,211 @@
         Result =  59
 """
 
-import operator as op
+# Defining valid unary operator symbols
+UNARY_OP_SYMBOLS = ("-", "+")
+
+# Defining valid binary operator symbols
+BINARY_OP_SYMBOLS = ("-", "+", "*", "^", "/")
+
+
+def get_number(data: str) -> int | float | str:
+    """
+    Converts the given data to appropriate number if it is indeed a number, else returns
+    the data as it is with a False flag. This function also serves as a check of whether
+    the input is a number or not.
+
+    Parameters
+    ----------
+    data : str
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    bool,  int or float
+        Returns a tuple of (a, b) where 'a' is True if data is indeed a number (integer
+        or numeric) and 'b' is either an integer of a floating point number.
+        If 'a' is False, then b is 'data'
+    """
+    try:
+        return int(data)
+    except ValueError:
+        try:
+            return float(data)
+        except ValueError:
+            if is_operator(data):
+                return data
+            msg = f"{data} is neither a number nor a valid operator"
+            raise ValueError(msg)
+
+
+def is_operator(data: str | int | float) -> bool:
+    """
+    Checks whether a given input is one of the valid operators or not.
+    Valid operators being '-', '+', '*', '^' and '/'.
+
+    Parameters
+    ----------
+    data : str
+        The value that needs to be checked for operator
+
+    Returns
+    -------
+    bool
+        True if data is an operator else False.
+    """
+    return data in BINARY_OP_SYMBOLS
+
+
+def evaluate(post_fix: list[str], verbose: bool = False) -> int | float | str | None:
```

Comment:
Yes, this would never return a string or None. But mypy test is failing because apparently it thinks the stack[0] (which is being returned) is made from both float and string as x is appended to the stack and x can be both float and string. See snippet below-
![image](https://github.com/TheAlgorithms/Python/assets/12524022/93810bbb-a42d-454f-b3ef-aaaa27c73901)
.
The mypy test passes when I cast `stack[0]` to float before returning. Shall I procced by casting it to float then?

Reply:
Yeah, I think casting to float before returning would be best.